lib/wowrb.rb

```diff
@@ -161,10 +161,24 @@ def self.credentials
   end
   private_class_method :credentials
 
+  def self.check
+    check = ENV['BATTLE_NET_KEY'].nil? || ENV['BATTLE_NET_REGION'].nil? || ENV['BATTLE_NET_LOCALE'].nil?
+    message =  'Env variables missing:' if check
+    message += " ENV['BATTLE_NET_KEY']" if ENV['BATTLE_NET_KEY'].nil?
+    message += " ENV['BATTLE_NET_REGION']" if ENV['BATTLE_NET_REGION'].nil?
+    message += " ENV['BATTLE_NET_LOCALE']" if ENV['BATTLE_NET_LOCALE'].nil?
+    check ? message : true
+  end
+
   def self.call_api(remote_url)
-    encoded_url = URI.encode(remote_url)
-    response = HTTParty.get(encoded_url)
-    JSON.parse(response.body)
+    if check == true
```

Comment:
Also it would be better to assign `check` method to variable and compare(and call in `else`) this variable